lib/flash_flow/data/collection.rb

```diff
@@ -65,6 +65,7 @@ def reverse_merge(old)
             branch.created_at = info.created_at
             branch.resolutions = info.resolutions.to_h.merge(branch.resolutions.to_h)
             branch.stories = info.stories.to_a | merged_branches[full_ref].stories.to_a
+            branch.merge_order ||= info.merge_order
```

Comment:
This doesn't seem right anymore since you're initializing the merge order with zero.